doc/rst/developer/bestPractices/ContributorInfo.rst

```diff
@@ -9,184 +9,274 @@ The git repository for the project is hosted on Github at
+   work is not being pursued redundantly, etc.  Similarly, fundamental changes
+   to the language or architecture should be circulated to the
+   chapel-developers_ and/or chapel-users_ lists to make sure effort is not
+   wasted.
+
+#. Sign a Chapel contributor's agreement and mail it, with your GitHub
+   ID, using the instructions here:
+
+   https://github.com/chapel-lang/chapel/tree/master/doc/rst/developer/contributorAgreements/
+
+* You do not need commit/push access to the main repo in order to
+  contribute code.  See `Who has commit access`_ for questions.
+
+* Third-party code requires additional approvals, see the policy details on
+  `Third-party code`_.
 
 .. _Create new branch:
 
 Create new branch
-~~~~~~~~~~~~~~~~~
++++++++++++++++++
 
-* Develop your feature, bug fix, etc on your fork. Using a concisely named
-  branch is encouraged.
+Develop your feature, bug fix, etc on your fork.  To create a new branch, use
+the `New branch command`_.  Using a concisely named branch is encouraged.
 
-.. _Submit pull request:
+.. _Develop and test contributions locally:
 
-Submit pull request
-~~~~~~~~~~~~~~~~~~~
+Develop and test contributions locally
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-* `Submit a pull request`_ with your changes (make sure you have `synced with
-  the main repo`_).
+Your contribution will take the form of a series of commits.  While including
+sensible commit messages is a good idea, it is more important to have a good
+merge message once the pull request is going in. Likewise, it is OK to have many
+small commits that reflect the history of development rather than commits for
+the feature.  See `Development commands`_ for how to perform some common
+operations during development.
 
-* In order for a pull request to be accepted and merged, it should a) pass
-  testing and b) be reviewed by a member of the core Chapel team (currently, a
-  member of the Cray Chapel team).
+As you work, you will want to periodically bring in changes from the main Chapel
+project to your feature branch (described in `Development commands`_), to avoid
+code drift.
 
-* It is considered good practice to keep PRs (pull requests) to a reasonable
-  size. This ensures that the PR will be reviewed in a timely manner and will
-  receive a higher level of attention per line of code during review.
+.. _Adding new tests:
 
-  * When submitting a PR, the contributor should ask themselves if their
-    contribution can be separated into smaller logical chunks or independent
-    parts. Reviewers will also be pondering the same question and may request a
-    break up of the contribution into smaller PRs.
+Adding new tests
+++++++++++++++++
 
-  * Breaking up a PR can sometimes require a great deal of effort and
-    creativity, and may not be feasible at all, due to the intertwined nature
-    of the code.
+You will probably need to create new tests for your feature. See `Test System`_
+for more information on this process.
```

Comment:
Can we also link to "Creating a Simple Test" in Test System? That's the information a beginner might immediately need.